Signal/src/AppDelegate.m

```diff
@@ -333,10 +334,39 @@ - (void)application:(UIApplication *)application
 - (BOOL)application:(UIApplication *)application continueUserActivity:(nonnull NSUserActivity *)userActivity restorationHandler:(nonnull void (^)(NSArray * _Nullable))restorationHandler
 {
     if ([userActivity.activityType isEqualToString:@"INStartVideoCallIntent"]) {
+        DDLogInfo(@"%@ got start video call intent", self.tag);
         [[Environment getCurrent].callService handleCallKitStartVideo];
+    } else if ([userActivity.activityType isEqualToString:@"INStartAudioCallIntent"]) {
+
+        if (!SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(10, 0)) {
+            DDLogError(@"%@ unexpectedly received INStartAudioCallIntent pre iOS10", self.tag);
+            return NO;
+        }
+
+        DDLogInfo(@"%@ got start audio call intent", self.tag);
+
+        INInteraction *interaction = [userActivity interaction];
+        INIntent *intent = interaction.intent;
+
+        if (![intent isKindOfClass:[INStartAudioCallIntent class]]) {
+            DDLogError(@"%@ unexpected class for start call audio: %@", self.tag, intent);
+            return NO;
+        }
+        INStartAudioCallIntent *startCallIntent = (INStartAudioCallIntent *)intent;
+        NSString *_Nullable handle = startCallIntent.contacts.firstObject.personHandle.value;
```

Comment:
It might be nice to add a comment here documenting what type of value `handle` is and how its formatted.